### PR TITLE
TILA-1765: Calculate reservation price on create/update

### PR DESCRIPTION
**Pull request by saaste**
The logic of the calculation is described in [TILA-1765](https://helsinkisolutionoffice.atlassian.net/browse/TILA-1765). When reservation is updated, price is calculated only if begin or end time is updated or when reservation units are changed.

### CODE REVIEW

File: api/graphql/reservations/reservation_serializers.py [hunk 1]
```diff
@@ -402,6 +425,92 @@ def check_reservation_kind(self, reservation_unit):
...
+        }
+
+        total_reservation_price: Decimal = Decimal(0.0)
+
+        first_paid_unit_price: Decimal = 0.0
+        first_paid_unit_tax_percentage: Decimal = 0.0
+        is_first_paid_set = False
+
+        for reservation_unit in reservation_units:
+            # If unit is not paid, there is no need for calculations. Skip.
```
Comment (nicobav) [inline api/graphql/reservations/reservation_serializers.py:475]: This wording can confuse: "If unit is not paid,". I suggest changing it to "If unit pricing type is not PAID,"

**Review by nicobav — CHANGES_REQUESTED** [review]

File: api/graphql/reservations/reservation_serializers.py [hunk 2]
```diff
@@ -40,6 +42,19 @@
 }
 
 
+class PriceCalculationResult:
+    reservation_price: Decimal = 0.0
```
[the remark this answers is not in the document]
Comment (saaste) [reply api/graphql/reservations/reservation_serializers.py]: Oh, crap! I'm still in statically typed world 😄 Fixed!

**Review by nicobav — APPROVED** [review]
LGTM :rocket:

File: api/graphql/reservations/reservation_serializers.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (nicobav) [reply api/graphql/reservations/reservation_serializers.py]: And I seem to stutter also when I write :sweat_smile:

[the remark this answers is not in the document]
Comment (nicobav) [reply api/graphql/reservations/reservation_serializers.py]: > though using the the type declaration

**Pull request merged**